src/tactic/core.lean

```diff
@@ -1612,6 +1612,43 @@ add_tactic_doc
 
 attribute [higher_order map_comp_pure] map_pure
 
+/-- Attribute `needed` is used to mark declarations that are featured
```

Comment:
Sorry, I don't find this explanation very clear. What does "featured" mean in this context? "Support the main features"? After reading the whole PR I think I get what it's trying to say, but it would be great if you could clarify this.

Reply:
The point is when building a new theory, we can often experiment with different formulations. By marking the parts we know we want to keep with `@[needed]`, `#unneeded` will help us identify the parts that are now obsolete. I can add this to the comment if that helps.